components/merino/src/merino.rs

```diff
@@ -0,0 +1,844 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+use serde::{Deserialize, Serialize};
+use url::Url;
+use uuid::Uuid;
+
+// Locales supported by Merino Curated Reccomendations
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+pub enum Locale {
+    #[serde(rename = "fr")]
+    Fr,
+    #[serde(rename = "fr-FR")]
+    FrFr,
+    #[serde(rename = "es")]
+    Es,
+    #[serde(rename = "es-ES")]
+    EsEs,
+    #[serde(rename = "it")]
+    It,
+    #[serde(rename = "it-IT")]
+    ItIt,
+    #[serde(rename = "en")]
+    En,
+    #[serde(rename = "en-CA")]
+    EnCa,
+    #[serde(rename = "en-GB")]
+    EnGb,
+    #[serde(rename = "en-US")]
+    EnUs,
+    #[serde(rename = "de")]
+    De,
+    #[serde(rename = "de-DE")]
+    DeDe,
+    #[serde(rename = "de-AT")]
+    DeAt,
+    #[serde(rename = "de-CH")]
+    DeCh,
+}
+// Configuration settings for a Section
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+pub struct SectionSettings {
+    #[serde(rename = "sectionId")]
+    section_id: String,
+    #[serde(rename = "isFollowed")]
+    is_followed: bool,
+    #[serde(rename = "isBlocked")]
+    is_blocked: bool,
+}
+
+// Information required to request curated recommendations
+#[derive(Debug, Serialize, Deserialize, PartialEq)]
+pub struct CuratedRecommendationsRequest {
+    pub locale: Locale,
+    pub region: Option<String>,
+    pub count: Option<i32>,
+    pub topics: Option<Vec<String>>,
+    pub feeds: Option<Vec<String>>,
+    pub sections: Option<Vec<SectionSettings>>,
+    #[serde(rename = "experimentName")]
+    pub experiment_name: Option<String>,
+    #[serde(rename = "experimentBranch")]
+    pub experiment_branch: Option<String>,
+}
```

Comment:
Is this going to be exposed to the consumer.  You might want to make this derive `uniffi::Record` and also add ` #[uniffi(default = None]` for the optional fields.  (https://mozilla.github.io/uniffi-rs/next/types/records.html).  This could be done in a second pass though.